src/crypto/pki.rs

```diff
@@ -2,17 +2,18 @@
 // SPDX-License-Identifier: AGPL-3.0-only
 
 use anyhow::{anyhow, bail, Result};
-use pkcs8::{ObjectIdentifier, PrivateKeyInfo, SubjectPublicKeyInfo};
+use pkcs8::{EncodePrivateKey, ObjectIdentifier, PrivateKeyInfo, SubjectPublicKeyInfo};
 use zeroize::Zeroizing;
 
-use der::{Decode, Encode};
+use der::Decode;
 use sec1::EcPrivateKey;
 use spki::AlgorithmIdentifier;
 
 use const_oid::db::rfc5912::{
     ECDSA_WITH_SHA_256, ECDSA_WITH_SHA_384, ID_EC_PUBLIC_KEY as ECPK, SECP_256_R_1 as P256,
     SECP_384_R_1 as P384,
 };
+use p256::ecdsa::signature::Signer;
```

Comment:
This could cause issues later if `p256` and `p384` diverge on the `ecdsa` crate version. So you should move this import down into the match statements (i.e. near lines 97 and 103).

Reply:
Done